patches/tModLoader/Terraria.ModLoader.Core/LogArchiver.cs

```diff
@@ -13,6 +13,20 @@ namespace Terraria.ModLoader.Core
 	internal static class LogArchiver
 	{
 		internal static void ArchiveLogs() {
+			if (!Directory.Exists(Logging.LogArchiveDir)) {
+				var pattern = new Regex(".*\\.zip");
+				var existingOldLogs = Directory.GetFiles(Logging.LogDir).Where(s => pattern.IsMatch(Path.GetFileName(s))).ToList();

```

Comment:
You don't need Regex nor Linq here. Just use Directory.GetFiles(Logging.LogDir, "*.zip"); here.

Reply:
I dont know why they are getting converted to a list too